scripts/partner_batch_imports.py

```diff
@@ -163,7 +167,7 @@ def batch_import(path, batch, batch_size=5000):
 
     for fname in filenames:
         book_items = []
-        with open(fname) as f:
+        with open(fname, 'r', encoding="ISO-8859-1") as f:
```

Comment:
@mekarpeles I believe the correct thing to do here, if there are problems with character encoding, is to read the file in binary mode, since the input is _supposed_ to be UTF8 encoded, but checking each line for encoding issues.

Trying to pick one encoding for a UTF8 file with a range of broken encodings isn't going to produce good results. 
ISO-8859-1 is guaranteed to be wrong, except on a handful of erroneous input data rows.

Reply:
This is an example of how to iterate through one of these file supposedly UTF8 encoded with some bad rows:

```
with open(sys.argv[1], 'rb') as f:
    while True:
        line = f.readline()
        if not line:
            break
        try:
            data = line.decode().strip().split('|')
        except UnicodeDecodeError:
            print('BAD LINE')
            continue

```

Reply:
I have to double check but I thought the behavior I was experiencing was that it literally failed to open the file. I'll double check & maybe pair w/ @jimchamp on Wednesday during 1:1

Reply:
@mekarpeles failing to open the file is consistent with what I observed with the code above. Opening the file in binary mode allows the encoding to be specified and checked for each line. I'm not aware of a permissive mode to open an entire file as a particular encoding but handle specific errors line by line. (If there is one, I'd be interested in hearing about it!)

My observation is that some lines are  ISO-8859-1 encoded, other lines are UTF8 encoded. The spec states the entire file should be UTF8 encoded, so the ISO-8859-1 encoded rows are errors. I haven't checked, but I'm assuming there are more correctly encoded UTF8 rows that erroneous ISO-8859-1 rows.

I would distrust any row in that data that is not correctly UTF8 encoded.

Reply:
I tried the code above and am seeing that ~47678 rows out of 876534 (~5%) fail, and they have title or author names like:

> St\xfctzle, Walther

If I decode the bytes to line.decode('ISO-8859-1'), I get
`Stützle, Walther`

Is it reasonable to use the code above and (on failing lines) retry with `line.decode('ISO-8859-1')`?

Reply:
@mekarpeles it sounds like it would work, but risks accepting other badly encoded data. Having these non-UTF-8 rows in the data in the first place is sloppy, which makes me suspicious of them in general, but using `line.decode('ISO-8859-1')` as a fall back on errors will squeeze out more records to import, if enough of those rows look good enough to import other than the encoding issue.